src/Events/Server/MiscEvents.cpp

```diff
@@ -167,4 +167,19 @@ REGISTER_LOCAL_EVENT(
 	}
 );
 
+REGISTER_LOCAL_EVENT(
+	alt::CEvent::Type::WINDOW_RESOLUTION_CHANGE,
+	windowResolutionChange,
+	[](LuaResourceImpl* resourceImpl, const alt::CEvent* ev) -> int
+	{
+		const alt::CWindowResolutionChangeEvent* event = static_cast<const alt::CWindowResolutionChangeEvent*>(ev);
+		lua_State* L = resourceImpl->GetLuaState();
+
+		lua_pushbaseobject(L, event->GetOldResolution());
```

Comment:
although it should work with `lua_pushbaseobject` you should use `lua_pushvector` instead because the difference between the two is that pushing a baseobject (`lua_pushbaseobject`) will reference the object in the lua registry so it won't be garbage collected, but with `lua_pushvector` it will not save reference in the registry so it can be garbage collected